### PR TITLE
[TM] Add spec for ExceptionsManager

### PR DESCRIPTION
**Pull request by thymikee**
## Summary

Part of #24875, adds a spec for ExceptionsManager

## Changelog

[General] [Added] - TM Add spec for ExceptionsManager

## Test Plan

Flow passes.

### CODE REVIEW

File: Libraries/Core/NativeExceptionsManager.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  ) => void;
+  +updateExceptionMessage: (
+    message: string,
+    stack: Array<StackFrame>,
+    exceptionId: number,
+  ) => void;
+  /**
+   * Available only on Android
+   */
+  +dismissRedbox: () => void;
```
Comment (thymikee) [inline Libraries/Core/NativeExceptionsManager.js]: what do we do about it @rickhanlonii?

Comment (zhongwuzw) [reply Libraries/Core/NativeExceptionsManager.js]: Should we need to split it to Android and iOS? if not, seems we need to label it for Codegen.

Comment (rickhanlonii) [reply Libraries/Core/NativeExceptionsManager.js]: For additional properties we have decided to generate it on both platforms and no-op on one, so for this you can update the comment to `// Android only`

Splitting out would be required for props/events that conflict

File: Libraries/Core/NativeExceptionsManager.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @flow strict-local
+ * @format
+ */
+
+'use strict';
+
+import type {TurboModule} from 'RCTExport';
+import * as TurboModuleRegistry from 'TurboModuleRegistry';
+import type {StackFrame} from './Devtools/parseErrorStack';
```
[the remark this answers is not in the document]
Comment (thymikee) [reply Libraries/Core/NativeExceptionsManager.js]: Moved the type def to NativeExceptionsManager as there wasn't too many call-sites to change

File: Libraries/Core/NativeExceptionsManager.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    stack: Array<StackFrame>,
+    exceptionId: number,
+  ) => void;
+  +updateExceptionMessage: (
+    message: string,
+    stack: Array<StackFrame>,
+    exceptionId: number,
+  ) => void;
+  // Android only
+  +dismissRedbox?: () => void;
```
Comment (zhongwuzw) [inline Libraries/Core/NativeExceptionsManager.js]: @RSNara  🤔 Seems `sentIntent` of `IntentAndroid` not marks optional
https://github.com/facebook/react-native/pull/24877/files#diff-7033e846872152e52446abb20da3efb8R25, should we also changed that?

Comment (RSNara) [reply Libraries/Core/NativeExceptionsManager.js]: Yeah, nice catch. We should change that.

Comment (RSNara) [reply Libraries/Core/NativeExceptionsManager.js]: Actually, no. I was wrong earlier about our codegen output. Turns out, regardless of if a method is optional or not, it'll generate a required method in native. So, making this optional doesn't do anything.

Comment (zhongwuzw) [reply Libraries/Core/NativeExceptionsManager.js]: 👌 Get it.

**Pull request closed without merging**